src/filedata.h

```diff
@@ -293,16 +278,66 @@ class FileData {
 
 	static void file_data_dump();
 
+    protected:
+	static FileData *file_data_new(const gchar *path_utf8, struct stat *st, gboolean disable_sidecars);
+	static FileData *file_data_new_local(const gchar *path, struct stat *st, gboolean disable_sidecars);
+
+	static GHashTable *file_data_basename_hash_new();
+	static GList *file_data_basename_hash_insert(GHashTable *basename_hash, FileData *fd);
+	static void file_data_basename_hash_insert_cb(gpointer fd, gpointer basename_hash);
+	static void file_data_basename_hash_remove_list(gpointer, gpointer value, gpointer);
+	static void file_data_basename_hash_free(GHashTable *basename_hash);
+	static void file_data_basename_hash_to_sidecars(gpointer, gpointer value, gpointer);
+
     private:
 	void set_exif_time_data_unused(GList *files);
 	void set_exif_time_digitized_data_unused(GList *files);
 	void set_rating_data_unused(GList *files);
-	GList *filelist_insert_sort_unused(GList *list, FileData *fd, SortType method, gboolean ascend);
 	gint file_data_get_user_orientation_unused(FileData *fd);
 	void file_data_set_user_orientation_unused(FileData *fd, gint value);
 	gboolean file_data_send_notification_idle_cb_unused(gpointer data);
 };
 
+class FileData::FileList
+{
+    private:
+	FileList() = delete;
+	friend class FileData;  // Allows FileData to access protected API.
+
+	// Globals.
+	static SortType sort_method;
+	static gboolean sort_ascend;
+	static gboolean sort_case;
+
+    public:
+	static gint sort_compare_filedata(FileData *fa, FileData *fb);
+	static gint sort_compare_filedata_full(FileData *fa, FileData *fb, SortType method, gboolean ascend);
```

Comment:
Not sure these methods belong to `FileList`. But since it's mechanical moving/renaming I'm ok for now. TODO note would be great.

Reply:
Yeah, I agree that it doesn't belong here.  I don't know exactly what order I'll be creating the subclasses in, but there will be a `util.cc` file which will collect a lot of these.

Reply:
Oh, I see what happened.  The implementation of those functions depends on the `filelist_sort_method` globals, which I had moved to this class.  I have an idea for how to manage those globals better, but it'll take more work.